src/error.rs

```diff
@@ -127,6 +137,35 @@ impl From<std::convert::Infallible> for Error {
     }
 }
 
+/// Error returned when max capacity of `HeaderMap` is exceeded
+pub struct MaxSizeReached {
```

Comment:
To keep consistency, this error type should probably be in `http::header`, since other errors related to a specific type live in the same module (for example, `http::status::InvalidStatusCode`).

Since none of the other error types have a public constructor, I'd remove `pub fn new()` as well.